src/actions/commands/fold.ts

```diff
@@ -72,3 +75,42 @@ class CommandOpenAllFoldsRecursively extends CommandFold {
   keys = ['z', 'O'];
   commandName = 'editor.unfoldRecursively';
 }
+
+@RegisterAction
+class AddFold extends BaseOperator {
+  override modes = [Mode.Normal, Mode.Visual];
+  keys = ['z', 'f'];
+
+  readonly commandName = 'editor.createFoldingRangeFromSelection';
+
+  public async run(vimState: VimState, start: Position, end: Position): Promise<void> {
+    const previousSelections = vimState.lastVisualSelection;  // keep in case of Normal mode
+    vimState.editor.selection = new vscode.Selection(start, end);
```

Comment:
Why's this necessary?

Reply:
do you mean the `vimState.editor.selection =`? If so, the command `editor.createFoldingRangeFromSelection` does just that: it create a fold from the selection. I haven't found any other way of doing that. I don't think there is other API for that. 
https://github.com/microsoft/vscode/blob/3edcbec59a15e6bec9538d37b9ac47702d90a3a4/src/vs/editor/contrib/folding/browser/folding.ts#L1081

Reply:
Ah, I missed that this was an operator - makes sense.